test/test.bats

```diff
@@ -85,3 +85,54 @@ execdirect() {
   assert_success
   assert_output "./printargs ./shebang.sh foo bar1"
 }
+
+@test "relative-exec" {
+  exec0 env env ./printargs foo bar1 bar2
+  assert_success
+  assert_output "./printargs foo bar1 bar2"
+}
+
+@test "path-based-exec" {
+  PATH="$PATH:/work" exec0 env env printargs foo bar1 bar2
+  assert_success
+  assert_output "printargs foo bar1 bar2"
+}
+
+@test "shebang-path" {
+  exec0 ./shebang-path.sh ./shebang-path.sh foo bar1
+  assert_success
+  #assert_output "./printargs ./shebang-path.sh foo bar1"
+  assert_output "./printargs /work/shebang-path.sh foo bar1"
+}
+
+@test "shebang-path-shell" {
+  exec0 ./shebang-path2.sh ./shebang-path2.sh foo bar1
+  assert_success
+  assert_output "./printargs foo bar1"
+}
+
+@test "shell-command-relative" {
+  if [ -n "$BINFMT_EMULATOR" ]; then
+    skip "prepend_workdir_if_relative is altering the behaviour for args when run under emulation"
+  fi
+
+  exec0 sh sh -c './shebang-path.sh foo bar1 bar2'
+  assert_success
+  assert_output "./printargs ./shebang-path.sh foo bar1 bar2"
+}
+
+@test "shell-command-relative-direct" {
+  if [ -n "$BINFMT_EMULATOR" ]; then
+    skip "prepend_workdir_if_relative is altering the behaviour for args when run under emulation"
```

Comment:
Calling out that under emulation these 2 tests behave differently to native, and it's as a result of the prepend_workdir_if_relative handling